include/power/power.h

```diff
@@ -64,6 +64,26 @@ extern unsigned char sys_pm_idle_exit_notify;
  * @{
  */
 
+/**
+ * Power managment notifier struct
+ *
+ * This struct contains callbacks that are called when the target enters and
+ * exits power states.
```

Comment:
We need documentation here, and/or at the register/unregister functions, that explains when the set of notifiers can be changed.  The code iterates over the list without any protection for the list changing as a result of these functions being invoked.

Reply:
thinking about it, that is trick, the list is accessed from ISR as well, what makes mutex nonviable. If I use a semaphore I can't wait if its unavailable. In this case if the semaphore is not available I can't access the list and consequently not send notifications. The other problem is that suspend is not only called from the idle thread, it can be directly called from the application. Ideas ?

Reply:
I don't see how this has been resolved, so unresolving it.

AFAICT these are invoked from `_sys_suspend` and `_sys_resume`.  Are those from interrupt context?  Do the notification callbacks have to be invoked from that context, or can/should they be pushed to a work queue?

Reply:
@pabigot Forgot to document it. I just protected the list of notifiers with a k_spin_lock to avoid race condition when accessing these structs. Will document that these callbacks can be called from an interruption context, at this point I don't know if that is problem, so I'll keep this behavior since that is how is currently done.

In the future if needed, I change it to a workqueue. This won't change the API anyway.